addons/purchase/models/purchase.py

```diff
@@ -167,8 +155,17 @@ def name_get(self):
     def create(self, vals):
         if vals.get('name', 'New') == 'New':
             vals['name'] = self.env['ir.sequence'].next_by_code('purchase.order') or '/'
+        if vals.get('date_planned'):
+            for line in vals.get('order_line'):
+                line[2]['date_planned'] = vals.get('date_planned')
```

Comment:
Not sure but just a review based on https://github.com/odoo/odoo/pull/28921/files#diff-39d0613187d36f16c541734e80925e9dR267 and https://github.com/odoo/odoo/pull/35058/files#diff-f3712147f4b25231caa371bd83265dbeR1509
Cc: @pedrobaeza